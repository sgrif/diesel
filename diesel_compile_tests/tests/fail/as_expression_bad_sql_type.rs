extern crate diesel;
use diesel::expression::AsExpression;

#[derive(AsExpression)]
#[diesel(sql_type)]
struct Lol;

#[derive(AsExpression)]
#[diesel(sql_type(Foo))]
struct Lol2;

#[derive(AsExpression)]
#[diesel(sql_type = "@%&&*")]
struct Lol3;

#[derive(AsExpression)]
#[diesel(sql_type = 1omg)]
struct Lol4;

fn main() {}
