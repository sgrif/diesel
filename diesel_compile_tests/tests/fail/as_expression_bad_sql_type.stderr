error: `sql_type` is either unexpected or has bad format
 --> $DIR/as_expression_bad_sql_type.rs:5:10
  |
5 | #[diesel(sql_type)]
  |          ^^^^^^^^

error: `sql_type` is either unexpected or has bad format
 --> $DIR/as_expression_bad_sql_type.rs:9:10
  |
9 | #[diesel(sql_type(Foo))]
  |          ^^^^^^^^

error: expected one of: `for`, parentheses, `fn`, `unsafe`, `extern`, identifier, `::`, `<`, square brackets, `*`, `&`, `!`, `impl`, `_`, lifetime
  --> $DIR/as_expression_bad_sql_type.rs:13:21
   |
13 | #[diesel(sql_type = "@%&&*")]
   |                     ^^^^^^^

error: expected one of: `for`, parentheses, `fn`, `unsafe`, `extern`, identifier, `::`, `<`, square brackets, `*`, `&`, `!`, `impl`, `_`, lifetime
  --> $DIR/as_expression_bad_sql_type.rs:17:21
   |
17 | #[diesel(sql_type = 1omg)]
   |                     ^^^^
